pkg/server/event_handler_test.go

```diff
@@ -521,10 +537,15 @@ func testToJSON(tb testing.TB, in any) []byte {
 	return b
 }
 
-type failProcessor struct{}
+type failProcessor struct {
+	returnErr error
+}
 
 func (p *failProcessor) Process(_ context.Context, m *structpb.Struct) error {
-	return fmt.Errorf("always fail")
+	if p.returnErr == nil {
+		return fmt.Errorf("always fail")
+	}
+	return pmaperrors.Wrap(p.returnErr)
 }
```

Comment:
I know it's not in this PR we added `failProcessor`. But we should just have one `testProcessor`, just like the `testMessenger`